paddle/fluid/imperative/layer.cc

```diff
@@ -336,11 +336,15 @@ void OpBase::InvokeBackwardHooks() {
   }
 }
 
-void OpBase::RegisterBackwardHooks(const py::object& callable) {
+void OpBase::RegisterBackwardHooks(const py::object& callable, bool front) {
```

Comment:
why need front?

Reply:
`trace.py` append the `release_op` as the first grad hook, and we can get the op object in other hooks, maybe we can store the hooks using `stack` instead of `vector` ?